x-pack/legacy/plugins/snapshot_restore/public/app/components/policy_form/policy_form.tsx

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { Fragment, useState } from 'react';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiSpacer,
+} from '@elastic/eui';
+import { SlmPolicyPayload } from '../../../../common/types';
+import { PolicyValidation, validatePolicy } from '../../services/validation';
+import { useAppDependencies } from '../../index';
+import { PolicyStepLogistics, PolicyStepSettings, PolicyStepReview } from './steps';
+import { PolicyNavigation } from './navigation';
+
+interface Props {
+  policy: SlmPolicyPayload;
+  indices: string[];
+  currentUrl: string;
+  isEditing?: boolean;
+  isSaving: boolean;
+  saveError?: React.ReactNode;
+  clearSaveError: () => void;
+  onCancel: () => void;
+  onSave: (policy: SlmPolicyPayload) => void;
+}
+
+export const PolicyForm: React.FunctionComponent<Props> = ({
+  policy: originalPolicy,
+  indices,
+  currentUrl,
+  isEditing,
+  isSaving,
+  saveError,
+  clearSaveError,
+  onSave,
+}) => {
+  const {
+    core: {
+      i18n: { FormattedMessage },
+    },
+  } = useAppDependencies();
+
+  // Step state
+  const [currentStep, setCurrentStep] = useState<number>(1);
+  const [maxCompletedStep, setMaxCompletedStep] = useState<number>(0);
+  const stepMap: { [key: number]: any } = {
+    1: PolicyStepLogistics,
+    2: PolicyStepSettings,
+    3: PolicyStepReview,
+  };
+  const CurrentStepForm = stepMap[currentStep];
+
+  // Policy state
+  const [policy, setPolicy] = useState<SlmPolicyPayload>({
+    ...originalPolicy,
+    config: {
+      ...(originalPolicy.config || {}),
+    },
+  });
+
+  // Policy validation state
+  const [validation, setValidation] = useState<PolicyValidation>({
+    isValid: true,
+    errors: {},
+  });
+
+  const updatePolicy = (updatedFields: any): void => {
+    const newPolicy = { ...policy, ...updatedFields };
+    const newValidation = validatePolicy(newPolicy);
+    setPolicy(newPolicy);
+    setValidation(newValidation);
+  };
+
+  const updateCurrentStep = (step: number) => {
+    if (maxCompletedStep < step - 1) {
+      return;
+    }
+    setCurrentStep(step);
+    setMaxCompletedStep(step - 1);
```

Comment:
> When in edit mode all steps should be enabled as the policy is valid (by doing previous point this will be done)

I agree, they should be enabled by default. I just realized that we want to reset the max completed step if you go back and invalidate a previous step. So your snippet would become this:

```js
const updateCurrentStep = (nextStep: number) => {
  if (!validation.isValid) {
    // Reset to the current step so they user can't skip ahead by clicking the steps nav.
    setMaxCompletedStep(nextStep - 1);
    return;
  }

  setCurrentStep(nextStep);
  setMaxCompletedStep(prev => Math.max(prev, nextStep - 1));
  clearSaveError();
};

const onBack = () => {
  updateCurrentStep(currentStep - 1);
};

const onNext = () => {
  updateCurrentStep(currentStep + 1);
};
```

Reply:
Based on the UX in the rollup job wizard, I think these are the rules the UX should follow:

1. If the current step has not been validated, disable all subsequent steps.
2. Once the current step has been validated and the user navigates to the next step, that becomes the max completed step.
3. If the user navigates back to a previously completed step, enable steps up to the max completed step.
4. If the user enters invalid input at this previously completed step, disable all subsequent steps.
5. Once the input has been re-validated, re-enable steps up to the max completed step.